[sound_play] local variable 'filesound' referenced before assignment

When I use `sound_play` with this launch file,

```
  <node pkg="sound_play" type="soundplay_node.py" name="sound_play_jp"
        if="$(arg launch_sound_play)" respawn="$(arg sound_play_respawn)"
        machine="$(arg sound_play_machine)">
    <remap from="robotsound" to="robotsound_jp"/>
    <remap from="sound_play" to="robotsound_jp"/>
    <env name="PATH" value="$(find voice_text)/bin:$(env PATH)" />
  </node>
```

I got the following error.

```
[ERROR] [1653451319.526417] [/sound_play_jp:rosout]: Exception in actionlib callback: local variable 'filesound' referenced before assignment
[INFO] [1653451319.527568] [/sound_play_jp:rosout]: Traceback (most recent call last):
  File "/home/fetch/ros/melodic/src/ros-drivers/audio_common/sound_play/scripts/soundplay_node.py", line 291, in execute_cb
    sound = self.select_sound(data)
  File "/home/fetch/ros/melodic/src/ros-drivers/audio_common/sound_play/scripts/soundplay_node.py", line 109, in select_sound
    sound = filesound
UnboundLocalError: local variable 'filesound' referenced before assignment
```

I am still not sure why this is happening, but I'll investigate and make a PR as soon as possible.

--- a/sound_play/scripts/soundplay_node.py
+++ b/sound_play/scripts/soundplay_node.py
@@ -106,7 +106,7 @@ class SoundPlayNode(object):
                             'volume for cached wave has changed,'
                             'resetting volume')
                         filesound.sound.set_property('volume', data.volume)
-                sound = filesound
+                sound = self.filesound[data.arg]
             else:
                 absfilename = os.path.join(
                     roslib.packages.get_pkg_dir(data.arg2), data.arg)
@@ -132,7 +132,7 @@ class SoundPlayNode(object):
                             'volume for cached wave has changed,'
                             'resetting volume')
                         filesound.sound.set_property('volume', data.volume)
-                sound = filesound
+                sound = self.filesound[absfilename]
         elif data.sound == SoundRequest.SAY:
             voice_key = data.arg + '---' + data.arg2
             if voice_key not in self.voicesounds.keys():

